src/main/java/com/rtbhouse/utils/avro/FastDeserializerGenerator.java

```diff
@@ -205,7 +214,21 @@ private void processRecord(JVar schemaVariable, JVar containerVariable, final Sc
                     break;
                 }
             }
-
+            if (!recordAction.getShouldRead()) {
+                return;
+            }
+            // seek through actionIterator also for default values
+            Set<String> fieldNamesSet = record.getFields().stream().map(Schema.Field::name).collect(Collectors.toSet());
```

Comment:
And what about aliased fields? The names won't match so there's a possibility that we will process the aliased field as a default one.

Reply:
Ok, the writer record schema comes with already aliased field names.